decentraland/unity-explorer
# chore: lock global assets when loading a scene

## What does this PR change?

- Introduces `SceneAssetLock`. Its responbility is to lock the global assets from starting if the scene we are standing at are not ready.
- This first iteration relies on `GatheGLTFSystem` for changing the readiness state. Therefore, this loading lock will only be affected when the scene is on the readiness queue. Ergo, when the loading screen is present.
- Moves LOD and Road instantiation to `PreRenderingGroup` so it doesnt consume budget framecap budget for scene assets
- Adds an out of range boolean to partition to avoid unwanted scene loading

## How to test the changes?



1. Teleport in genesis plaza to differente locations
2. Teleport between realms
3. Move around genesis plaza
4. Ask someone to connect with you and teleport together
5. Teleport away from genesis plaza, and back. When you go back, the scene should be loaded. Not the LOD. You can confirm this with the lod debugger, or checking if the scene you are at is alive
6. Everything should work as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
